### ISSUE TITLE
Adding product attribute options through api can fail because of incorrect string comparison

### ISSUE THREAD

**Issue by koenner01** [magento/magento2#13151]
Trying to insert a new product attribute option through the api that contains an existing value but starting with a character '0' will fail. This also happens when directly using the Magento\Catalog\Api\ProductAttributeOptionManagementInterface add function

### Preconditions
1. PHP7
2. CE 2.2.2 (looking at the codebase on github this issue is present since the start of Magento2), also tested on EE 2.2.2

### Steps to reproduce
1. Create a product attribute option '95' manually for a random attribute
2. Try to insert a new option for the same random attribute through api '../rest/all/V1/products/attributes/{randomAttributeCode}/options' with body:
```
{
    "option": {
        "label": "095"
    }
}
```
3. Watch Magento fail time and time again in the add function of Magento\Catalog\Model\Product\Attribute\OptionManagement where an in_array check is done that causes false to be returned.

### Expected result
1. A new option '095' should be created

### Actual result
1. Api returns status 200 with response 'false'

Code snippet from Magento\Catalog\Model\Product\Attribute\OptionManagement add function:

```
public function add($attributeCode, $option)
{
    /** @var \Magento\Eav\Api\Data\AttributeOptionInterface[] $currentOptions */
    $currentOptions = $this->getItems($attributeCode);
    if (is_array($currentOptions)) {
        array_walk($currentOptions, function (&$attributeOption) {
            /** @var \Magento\Eav\Api\Data\AttributeOptionInterface $attributeOption */
                $attributeOption = $attributeOption->getLabel();
        });
        if (in_array($option->getLabel(), $currentOptions)) {
            return false;
        }
    }
    return $this->eavOptionManagement->add(
        \Magento\Catalog\Api\Data\ProductAttributeInterface::ENTITY_TYPE_CODE,
        $attributeCode,
        $option
    );
}
```

The in_array function is not working correctly; PHP documentation states that there is a third parameter that can be passed in the in_array function for strict validation. Adding this solves the problem

#magentomagic

**Comment by magento-engcom-team** [comment]
@koenner01, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by pmclain** [comment]
This is fixed in 2.3-develop by https://github.com/magento/magento2/pull/19451

**Comment by ihor-sviziev** [comment]
Hi @koenner01. Thank you for your report. 
The issue has been fixed in magento/magento2#19451 by @SikailoISM in 2.3-develop branch
Related commit(s):
 - [7e55ad5e4b2154110a66a55a41c870cfab44220f](https://github.com/magento/magento2/commit/7e55ad5e4b2154110a66a55a41c870cfab44220f)

The fix will be available with the upcoming 2.3.1 release.

**Comment by ihor-sviziev** [comment]
Hi @koenner01. Thank you for your report. 
The issue has been fixed in magento/magento2#19612 by @SikailoISM in 2.2-develop branch
Related commit(s):
 - [14aa091388aaea80195d874e47d874f8df6620d7](https://github.com/magento/magento2/commit/14aa091388aaea80195d874e47d874f8df6620d7)

The fix will be available with the upcoming 2.2.8 release.

**Comment by ihor-sviziev** [comment]
Duplicate of #19436
